tests/ga/test_remoteaccess_handler.py

```diff
@@ -0,0 +1,358 @@
+    return_value="]aPPEv}uNg1FPnl?")
+    def test_handle_remote_access_multiple_users(self, _):
+        rah = RemoteAccessHandler()
+        rah.os_util = MockOSUtil()
+        data_str = load_data('wire/remote_access_two_accounts.xml')
+        remote_access = RemoteAccess(data_str)
+        tstuser = remote_access.user_list.users[0].name
+        tstuser2 = remote_access.user_list.users[1].name
+        expiration_date = datetime.utcnow() + timedelta(days=1)
+        expiration_date2 = datetime.utcnow() + timedelta(days=5)
+        expiration = (expiration_date).strftime("%a, %d %b %Y %H:%M:%S ") + "UTC"
+        expiration2 = (expiration_date2).strftime("%a, %d %b %Y %H:%M:%S ") + "UTC"
+        remote_access.user_list.users[0].expiration = expiration
+        remote_access.user_list.users[1].expiration = expiration2
+        rah.remote_access = remote_access
+        rah.handle_remote_access()
+        os_util = rah.os_util
+        os_util.__class__ = MockOSUtil
+        self.assertTrue(tstuser, "{0} missing from users".format(tstuser))
+        self.assertTrue(tstuser2, "{0} missing from users".format(tstuser2))
+
+    @patch('azurelinuxagent.common.utils.cryptutil.CryptUtil.decrypt_secret',
+    return_value="]aPPEv}uNg1FPnl?")
+    # max fabric supports in the Goal State
+    def test_handle_remote_access_ten_users(self, _):
+        rah = RemoteAccessHandler()
+        rah.os_util = MockOSUtil()
+        data_str = load_data('wire/remote_access_10_accounts.xml')
+        remote_access = RemoteAccess(data_str)
+        count = 0
+        for user in remote_access.user_list.users:
+            count += 1
+            user.name = "tstuser{0}".format(count)
+            expiration_date = datetime.utcnow() + timedelta(days=count)
+            user.expiration = expiration_date.strftime("%a, %d %b %Y %H:%M:%S ") + "UTC"
+        rah.remote_access = remote_access
+        rah.handle_remote_access()
+        os_util = rah.os_util
+        os_util.__class__ = MockOSUtil
+        self.assertEqual(10, os_util.all_users.keys().__len__())
+
+    @patch('azurelinuxagent.common.utils.cryptutil.CryptUtil.decrypt_secret',
+    return_value="]aPPEv}uNg1FPnl?")
+    def test_handle_remote_access_user_removed(self, _):
+        rah = RemoteAccessHandler()
+        rah.os_util = MockOSUtil()
+        data_str = load_data('wire/remote_access_10_accounts.xml')
+        remote_access = RemoteAccess(data_str)
+        count = 0
+        for user in remote_access.user_list.users:
+            count += 1
+            user.name = "tstuser{0}".format(count)
+            expiration_date = datetime.utcnow() + timedelta(days=count)
+            user.expiration = expiration_date.strftime("%a, %d %b %Y %H:%M:%S ") + "UTC"
+        rah.remote_access = remote_access
+        rah.handle_remote_access()
+        os_util = rah.os_util
+        os_util.__class__ = MockOSUtil
+        self.assertEqual(10, os_util.all_users.keys().__len__())
```

Comment:
`os_util.all_users.keys().__len__()` should just be `len(os_util.all_users)`. You have to have an exceptionally good reason to ever invoke `__len__()` directly.

Reply:
Fixed.

---
In reply to: [194531437](https://github.com/Azure/WALinuxAgent/pull/1171#discussion_r194531437) [](ancestors = 194531437)